### DISCUSSION TITLE
WebUI Contribution Standard

### DISCUSSION THREAD

**Discussion by hassandraga** [webui-dev/webui#277] [General]
# WebUI Contribution Standard

## Binaries

1. All wrappers should NOT include any binary files in the repository in all cases
2. All wrappers should USE [bootstrap.sh](https://github.com/webui-dev/deno-webui/blob/main/src/bootstrap.sh) and [bootstrap.bat](https://github.com/webui-dev/deno-webui/blob/main/src/bootstrap.bat) in the `readme` to download the trusted binaries
3. All wrappers should INCLUDE the trusted binaries in the official online release package installer using those methods:
a. Method 1: Using [GitHub Action](https://github.com/webui-dev/python-webui/blob/main/.github/workflows/pypi-publish.yml) to download and inject the trusted binaries in the package, then publish it
b. Method 2: The wrapper runs the `bootstrap.sh` and `bootstrap.bat` at runtimes to download the trusted binaries
c. Method 3: The wrapper uses the available built-in APIs to download the trusted binaries `.zip` and extract it

## Code

1. No global code style is mandatory on the wrappers because each wrapper is written in a different language, so there are different needs. But in general, a working, well-organized code is welcomed
2. All wrappers should TRY to have as similar API names as possible
a. For example, `webui_exit` should NOT be implemented as `.exitApp` or `.ExitProgram`, but instead, it should be `.exit` or `.Exit`
b. All global APIs like `.exit` should be under the `webui.xx` object, while window APIs should be under `myWindow.xx` object
c. Any extra feature added to the wrapper should have its OWN dedicated API. For example, WebUI does not support JSON at all, so a wrapper with JSON support should be `.scriptJSON` or any similar name while keeping `.script` pure as the original WebUI core API

## Examples

1. The examples should NOT use short/optimized code like `w` but should use complete keyword names like `myWindow` or similar to help the new users understand the code. Examples are not a production release product but a showcase of how WebUI works
6. examples should HAVE comments as possible

## Documentation

1. All documentation should be unified on [one page](https://webui.me/docs/2.4/#/)

## License

1. All wrappers SHOULD be licensed under `MIT License`

**Comment by Frued26** [comment]
Thank you very much this.
